src/Mvc/Mvc.NewtonsoftJson/src/DependencyInjection/NewtonsoftJsonMvcOptionsSetup.cs

```diff
@@ -88,5 +91,10 @@ public void Configure(MvcOptions options)
 
         options.ModelMetadataDetailsProviders.Add(new SuppressChildValidationMetadataProvider(typeof(IJsonPatchDocument)));
         options.ModelMetadataDetailsProviders.Add(new SuppressChildValidationMetadataProvider(typeof(JToken)));
+
+        if (_jsonOptions.RespectModelNameInValidationErrors)
```

Comment:
It's a bit weird to try and read an option as part of the options setup. For instance, it'll fail for scenarios like this:

```C#
services.AddNewtonsoftJson();
services.Configure<MvcNewtonsoftJsonOptions>(options => options.RespectModelNameInValidationErrors = true); // This will be ignored.
```

You could consider using a PostConfigureOptions to workaround it. An alternative would be to make the metadata provider public and configured by default and users that want to opt out could remove it. Note that you couldn't have it on by default for System.Text.Json though since it's on by default, and would affect non-json input which we don't want. Maybe give people access to the details provider and have them add it if they'd like to?

Reply:
Interesting, let me re-think about it.  Do you think the provider is the best way to achieve it? I am asking because I tried to scope to only json inputs and I was not able to.

Reply:
I have been experimenting with your idea and I really like it. Do you see any problem with different default behaviors between STJ and NewtonsoftJson, since one will have the provider as default?

Reply:
@pranavkm , I just updated with your suggestion.

Reply:
Also updated the API suggestion #39010